doc/matching-content-to-subscriber-lists.md

```diff
@@ -0,0 +1,83 @@
+# Matching content to subscriber lists
+
+## Format of matching criteria
+
+A subscriber list uses several fields to store the criteria which define how it
+matches published content:
+
+### JSON fields
+
+- `links`
+- `tags`
+
+Only one of these should be used on a given subscriber list; `tags` is deprecated.
+They can also both be empty `{}` if they aren't needed.
+
+These fields contain a JSON object which maps keys to arrays of values, similar
+to the links hash on a content item. For example:
+
+```
+{
+    "key_1": ["value_a", "value_b", "value_c"],
+    "key_2": ["value_x", "value_y"]
+}
+```
+
+For a published content item to match the subscriber list on these fields, the
+links/tags in the request to email-alert-api must include all of the keys and at
+least one value for each key.
+
+`links` uses content ids and `tags` uses slugs - but the queries used on these
+fields don't care about that difference.
+
+It's crucial that the key names stored in these JSON fields match the key names
+used in the requests to email-alert-api (which are usually the same as the ones
+on the content item itself). It's easy to break existing subscriptions by
+renaming a field in the links hash on published content without updating the
+corresponding subscriber lists, which would mean that subscribers stop getting
+emails for that kind of content.
```

Comment:
but be worth mentioning `taxon_tree` which is a special case field that is generated in `email_alert_service` and not `publishing_api`